board.rb

```diff
@@ -15,45 +18,93 @@ def graphic_table
     counter = 0
     element = division = ""
 
-    for i in (0...@size)
-      for j in (0...@size)
-      
-        if @table[i][j].nil?
-          element += '-' + '|'
-        else
-          element += @table[i][j].to_s + '|'
-        end
-        counter += 1
-      end
+    @table.each do |rows|
+      element += rows.map{ |val|  val.nil? ? '-|' : "#{val}|" }.join
       element += "\n"
-      division = '-' * (@size * 2) + "\n" if (counter % @size).zero?
+      division = "#{'-' * (@size * 2)} \n" if (counter % @size).zero?
       element += division
     end
 
     element
   end
 
   def fill_table(position, symbol)
-    arr_pos = position_coverter position
-    if empty_position? arr_pos
-      @table[arr_pos[0]][arr_pos[1]] = symbol
-      return true
-    else
-      return false
-    end
+    write_cell(position, symbol) if empty_position? position
   end
 
   def table_guide
     table_str = ""
     (1..@size**2).each do |num|
-      table_str += "#{num}" + ' ' * ((@size * @size).to_s.length - num.to_s.length) + '|'
+      table_str += "#{num}#{ ' ' * ((@size * @size).to_s.length - num.to_s.length) }|"
```

Comment:
Hi @heyapricot the purpose of this code is that the table guide always looks square. For a 3x3 matrix you won't see the effect because we have only one digit on each cell but when the matrix is bigger (for example, two digits in some cells) we want that the cells with just one digit have the same space as the cell with two digits.

Reply:
I understand, but do you see the problem of calling `to_s` and then `length`? It returns the length of the number converted to a string. E.g. `100 * 100` = `10000` `10000.to_s` = `'10000'` `'10000'.length` = `5`

Is that what you are trying to achieve? I don't think so

Reply:
That is what I'm trying to achieve because I need to count how many digits the number has to know how many spaces I'm going to put there and if I use the integer instead of the string (`1000.length`) I will get an error because `length` is not defined for numeric values.